src/postfork.cpp

```diff
@@ -60,36 +60,16 @@ static void debug_safe_int(int level, const char *format, int val) {
 }
 
 /// Called only by the child to set its own process group (possibly creating a new group in the
-/// process if it is the first in a JOB_CONTROL job. The parent will wait for this to finish.
-/// A process that isn't already in control of the terminal can't give itself control of the
-/// terminal without hanging, but it's not right for the child to try and give itself control
-/// from the very beginning because the parent may not have gotten around to doing so yet. Let
-/// the parent figure it out; if the child doesn't have terminal control and it later tries to
-/// read from the terminal, the kernel will send it SIGTTIN and it'll hang anyway.
-/// The key here is that the parent should transfer control of the terminal (if appropriate)
-/// prior to sending the child SIGCONT to wake it up to exec.
-///
+/// process if it is the first in a JOB_CONTROL job.
 /// Returns true on sucess, false on failiure.
 bool child_set_group(job_t *j, process_t *p) {
     bool retval = true;
-
     if (j->get_flag(JOB_CONTROL)) {
         // New jobs have the pgid set to -2
         if (j->pgid == -2) {
             j->pgid = p->pid;
         }
-        // Retry on EPERM because there's no way that a child cannot join an existing progress group
-        // because we are SIGSTOPing the previous job in the chain. Sometimes we have to try a few
-        // times to get the kernel to see the new group. (Linux 4.4.0)
-        int failure = setpgid(p->pid, j->pgid);
-        while (failure == -1 && (errno == EPERM || errno == EINTR)) {
-            debug_safe(4, "Retrying setpgid in child process");
-            failure = setpgid(p->pid, j->pgid);
-        }
-        // TODO: Figure out why we're testing whether the pgid is correct after attempting to
-        // set it failed. This was added in commit 4e912ef8 from 2012-02-27.
-        failure = failure && getpgid(p->pid) != j->pgid;
-        if (failure) {  //!OCLINT(collapsible if statements)
+        if (setpgid(p->pid, j->pgid) < 0) {
```

Comment:
Just for reference and in case @ridiculousfish has any feedback on its initial purpose: while the `EPERM` check was added by myself to get around the WSL issue, the `EINTR` retry logic was from before. To me, `EINTR` retries make sense in blocking calls that might take a while to execute (such as `waitpid`) but for one-off calls like this I think littering `EINTR` checks everywhere is something we can do without so this should be ok?

Reply:
Yep, this got removed in a later commit.